pkg/rehearse/metrics.go

```diff
@@ -0,0 +1,131 @@
+package rehearse
+
+import (
+	"encoding/json"
+	"github.com/openshift/ci-operator-prowgen/pkg/config"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowconfig "k8s.io/test-infra/prow/config"
+)
+
+type ExecutionMetrics struct {
+	SubmittedRehearsals sets.String
+	FailedRehearsals    sets.String
+	PassedRehearsals    sets.String
+}
+
+func (m *ExecutionMetrics) RecordFailedRehearsal(job *v1.ProwJob) {
+	if m.FailedRehearsals == nil {
+		m.FailedRehearsals = sets.NewString()
+	}
+	m.FailedRehearsals.Insert(job.Spec.Job)
+}
+
+func (m *ExecutionMetrics) RecordPassedRehearsal(job *v1.ProwJob) {
+	if m.PassedRehearsals == nil {
+		m.PassedRehearsals = sets.NewString()
```

Comment:
I would prefer doing that inside a constructor, and void doing that check in the methods.

Reply:
Aye